store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java

```diff
@@ -0,0 +1,162 @@
+                                                                      @PathVariable(value = "publicKey") final String publicKey,
+                                                                      @Parameter(description = "The number of the page to obtain, starting from 1")
+                                                                      @RequestParam(value = "page", defaultValue = "1", required = false) final int page,
+                                                                      @Parameter(description = "The number of validator rewards to retrieved in a page, defaults to 10")
+                                                                      @RequestParam(value = "size", defaultValue = "10", required = false) final int size,
+                                                                      @Parameter(description = "The name of the field to sort on")
+                                                                      @RequestParam(value = "order_by", defaultValue = TIMESTAMP, required = false) final ValidationRewardSortableFields orderBy,
+                                                                      @Parameter(description = "The direction of the sort")
+                                                                      @RequestParam(value = "order_direction", defaultValue = "DESC", required = false) final Sort.Direction orderDirection) throws NoSuchAlgorithmException {
+
+        logger.debug("getValidatorRewards publicKey {}, page {}, size {}, orderBy {}, orderDirection {}",
+                publicKey,
+                page,
+                size,
+                orderBy,
+                orderDirection
+        );
+
+        var request = PageRequest.of(page - 1, size, getSort(orderBy, orderDirection));
+
+        return ResponseEntity.ok(new PageResponse<>(validatorRewardRepository.findByPublicKey(
+                PublicKey.fromTaggedHexString(publicKey), request)
+        ));
+    }
+
+    /**
+     * Obtains the total rewards of a validator.
+     *
+     * @param publicKey the validators public key
+     * @return the total rewards
+     * @throws NoSuchAlgorithmException on invalid key
+     */
+    @GetMapping(value = "/validators/{publicKey}/total-rewards", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(tags = "validator-total-rewards'", summary = "Obtains the validator's total rewards",
+            description = "Obtains the validator's total rewards")
+    ResponseEntity<BigInteger> getTotalValidatorRewards(@Parameter(description = "The public key of the validator whose total rewards are to be obtained")
+                                                        @PathVariable(value = "publicKey") final String publicKey) throws NoSuchAlgorithmException {
+
+        logger.debug("getTotalValidatorRewards publicKey {}", publicKey);
+
```

Comment:
validate the public key

Reply:
Will be done in https://github.com/stormeye2000/cspr-events-store/issues/23